web/packages/teleport/src/teleportContext.tsx

```diff
@@ -150,12 +150,21 @@ class TeleportContext implements types.Context {
       roles: userContext.getRoleAccess().list,
       trustedClusters: userContext.getTrustedClusterAccess().list,
       users: userContext.getUserAccess().list,
-      applications: userContext.getAppServerAccess().list,
-      kubernetes: userContext.getKubeServerAccess().list,
+      applications:
+        userContext.getAppServerAccess().list &&
+        userContext.getAppServerAccess().read,
+      kubernetes:
+        userContext.getKubeServerAccess().list &&
+        userContext.getKubeServerAccess().read,
       billing: userContext.getBillingAccess().list,
-      databases: userContext.getDatabaseServerAccess().list,
-      desktops: userContext.getDesktopAccess().list,
-      nodes: userContext.getNodeAccess().list,
+      databases:
+        userContext.getDatabaseServerAccess().list &&
+        userContext.getDatabaseServerAccess().read,
+      desktops:
+        userContext.getDesktopAccess().list &&
+        userContext.getDesktopAccess().read,
+      nodes:
```

Comment:
you should double check for the others, but i know nodes, only does `list` check in the back

Reply:
looks like only nodes does `list` and everyting else is both. thank you